src/modules/main/sections/MainWrapper.vue

```diff
@@ -84,16 +85,26 @@ export default {
 	},
 
 	methods: {
+		setActiveElement() {
```

Comment:
I'm not sure if having a global state for the current active element is not causing us troubles later if we consider asynchronous operations. Thinking about a request that is performed and in the meantime the user interacts with another view or table already which changes the active id.

What do you think about just using the tableId or viewId as a parameter in the events?

For example:

```
emit('tables:sidebar:sharing', { viewId: this.element.id, open: true, tab: 'sharing' })
```

Or even more flexible something like:

```
emit('tables:sidebar:sharing', { elementType: this.isView ? 'view' : 'table', elementId: this.element.id, open: true, tab: 'sharing' })
```

Reply:
Yep, I didn't consider asynchronous operations. Good catch.